### PR TITLE
removed unused cassettes. added code to check if cassettes are unused

### PR DESCRIPTION
**Pull request by vpereira**
we have some superseded cassettes laying around. I implemented a strategy to after run all the suite, the unused cassettes, are deleted.

It works well, with this commit - I removed some cassettes, however I saw that if I run just one spec file, it will remove all "unused" - it means all others cassettes, which is wrong. so It should be fixed. I thought that the after(:suite) block would run just with the whole suite, but looks like I'm wrong

### CODE REVIEW

**Comment by coolo** [comment]
If we go with #5212 you can just remove all cassettes with an old timestamp

File: src/api/spec/rails_helper.rb [hunk 1]
```diff
@@ -19,6 +19,21 @@
 # check for pending migration and apply them before tests are run.
 ActiveRecord::Migration.maintain_test_schema!
 
+require 'set'
```
Comment (ChrisBr) [inline src/api/spec/rails_helper.rb]: This should go into a dedicated file!

**Comment by ChrisBr** [comment]
I like the idea :+1: 

Regarding your problem with running only a few specs. I think it is unlikely running the whole test suite on your local machine (I don't remember when I did it). So I would run this only in CI. Something like

```
if ENV['CIRCLECI']
...  
end
```

should do the trick. I also wouldn't delete the files but instead either raise an exception to fail the test suite and print out the cassettes which the dev needs to delete. Or not that invasive / strict just printing a warning message. I would prefer the first approach! After merging this PR we should be in a clean state and everytime somebody changes the specs, CI will fail when cassettes are obsolete.

**Comment by coolo** [comment]
This will make it impossible to split the rspec testsuite

**Comment by coolo** [comment]
(as in parallalize)

**Comment by ChrisBr** [comment]
> This will make it impossible to split the rspec testsuite

Good point @coolo :+1:

**Comment by ChrisBr** [comment]
> If we go with #5212 you can just remove all cassettes with an old timestamp

What is the state of #5212 ? I didn't follow the conversation between you and @hennevogel

**Comment by coolo** [comment]
For one, I'm waiting on https://github.com/openSUSE/open-build-service/pull/5293 to rebase on it

File: src/api/spec/rails_helper.rb (hunk 1 above)
Comment (hennevogel) [reply src/api/spec/rails_helper.rb]: All of this should

**Comment by hennevogel** [comment]
Extending an internal API from `VCR` for some unused files that don't do any harm. Good luck to your colleagues in 2 years when this breaks and you are not there anymore...

```
¯\_(ツ)_/¯
```

**Comment by vpereira** [comment]
> Extending an internal API from VCR for some unused files that don't do any harm
@hennevogel  some 108 unused files and the number just tends to grow. So to have a way to delete it is a good way 

> Good luck to your colleagues in 2 years when this breaks and you are not there anymore...
I hope in two years we are already away from VCR :)

**Comment by hennevogel** [comment]
>  some 108 unused files and the number just tends to grow.

Don't get me wrong, I'm not against removing those files. But why do I need this code of I can just remove all files and run the test suite again?

> I hope in two years we are already away from VCR :)

Hope dies last :skull:  Are you sure you want to introduce code based on hope?

**Comment by vpereira** [comment]
The code isn't introduced based on hope but based on the upstream
suggestion https://github.com/vcr/vcr/issues/283. Feel free to close the PR
or suggest a rework if necessary

**Comment by vpereira** [comment]
why is the unrelated test on circleci breaking but not on travis?

**Pull request merged**